django/contrib/admin/static/admin/js/change_form.js

```diff
@@ -2,8 +2,35 @@
 {
     const inputTags = ['BUTTON', 'INPUT', 'SELECT', 'TEXTAREA'];
     const modelName = document.getElementById('django-admin-form-add-constants').dataset.modelName;
+    const buttons = document.querySelectorAll('input[type=submit]');
```

Comment:
If the user extends the admin with extra forms somehow, this query selector will capture those forms’ submit buttons. But later the submit event listener is specific to the one form. I think it's better to do this later, under `if (modelName)`, and find only submit buttons in the form with `form.querySelectorAll(...)`

Reply:
Oh, good catch!

Done!